src/actions/create-environment/index.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as core from '@actions/core';
+import { DefaultRunnerFactory, RunnerFactory } from '../../lib';
+
+(async () => {
+    if (process.env.GITHUB_ACTIONS) {
+        await main(DefaultRunnerFactory);
+    }
+})();
+
+export async function main(factory: RunnerFactory): Promise<void> {
+    try {
+        core.startGroup('create-environment');
+        const envUrl = core.getInput('environment-url', { required: true });
+        const username = core.getInput('user-name', { required: true });
+        core.info(`environmentUrl: ${envUrl}; login as user: ${username}`);
+
+        const password = core.getInput('password-secret', { required: true });
+        if (!password || password.length === 0) {
+            return core.setFailed('Missing password! Specify one by setting input: \'password-secret\'');
+        }
+
+        const envName = core.getInput('name', { required: true});
+        const envRegion = core.getInput('region', {required: true});
+        const envType = core.getInput('type', {required: true});
+
+        const pac = factory.getRunner('pac', process.cwd());
+        await pac.run(['auth', 'create', '--url', envUrl, '--username', username, '--password', password]);
```

Comment:
that code won't work: for CRUD operations, this needs to be a pac auth create --kind ADMIN